azurerm/internal/services/advisor/resource_arm_advisor_suppression.go

```diff
@@ -0,0 +1,162 @@
+package advisor
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/advisor/mgmt/2020-01-01/advisor"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/advisor/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/advisor/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAdvisorSuppression() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAdvisorSuppressionCreateUpdate,
+		Read:   resourceArmAdvisorSuppressionRead,
+		Update: resourceArmAdvisorSuppressionCreateUpdate,
+		Delete: resourceArmAdvisorSuppressionDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.AdvisorSuppressionID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.AdvisorSuppressionName(),
+			},
+
+			"recommendation_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.AdvisorRecommendationID,
+			},
+
+			"duration_days": {
```

Comment:
could we make this 
```suggestion
			"duration_of_days": {
```
or
```suggestion
			"duration_in_days": {
```